### PR TITLE
Feature/sora staking rewards

### PR DESCRIPTION
**Pull request by bnsports**

### CODE REVIEW

**Review by Viscount88 — APPROVED** [review]

File: fearless/Common/Services/PayoutRewardsService/SoraSubsquidPayoutValidatorForNominatorFactory.swift [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        self.url = url
+        self.chainAsset = chainAsset
+    }
+
+    private func createRequestFactory(
+        address: AccountAddress,
+        historyRange: @escaping () -> EraRange?
+    ) -> NetworkRequestFactoryProtocol {
+        BlockNetworkRequestFactory {
+            var request = URLRequest(url: self.url)
```
Comment (DRadmir) [inline fearless/Common/Services/PayoutRewardsService/SoraSubsquidPayoutValidatorForNominatorFactory.swift:21]: use weak self

File: fearless/Common/Services/PayoutRewardsService/ValidatorPayoutInfoFactory.swift [hunk 2]
```diff
@@ -3,12 +3,10 @@ import IrohaCrypto
 import SSFModels
 
 final class ValidatorPayoutInfoFactory: PayoutInfoFactoryProtocol {
-    let chain: ChainModel
-    let asset: AssetModel
+    let chainAsset: ChainAsset
```
Comment (DRadmir) [inline fearless/Common/Services/PayoutRewardsService/ValidatorPayoutInfoFactory.swift:6]: private

File: fearless/CoreLayer/OperationFactory/BlockExplorer/Rewards/SoraRewardOperationFactory.swift [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+                throw CommonError.internal
+            }
+
+            let queryString = strongSelf.prepareHistoryRequestForAddress(
+                address,
+                startTimestamp: startTimestamp,
+                endTimestamp: endTimestamp
+            )
+
+            guard let url = self?.url else {
```
Comment (DRadmir) [inline fearless/CoreLayer/OperationFactory/BlockExplorer/Rewards/SoraRewardOperationFactory.swift:96]: Use strongSelf

File: fearless/CoreLayer/OperationFactory/BlockExplorer/Rewards/SoraRewardOperationFactory.swift [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+                forHTTPHeaderField: HttpHeaderKey.contentType.rawValue
+            )
+
+            request.httpMethod = HttpMethod.post.rawValue
+            return request
+        }
+
+        let resultFactory = AnyNetworkResultFactory<RewardOrSlashResponse> { data in
+            let response = try JSONDecoder().decode(
+                SubqueryResponse<GiantsquidResponseData>.self,
```
Comment (DRadmir) [inline fearless/CoreLayer/OperationFactory/BlockExplorer/Rewards/SoraRewardOperationFactory.swift:115]: Create response for Giantsquid

**Review by DRadmir — CHANGES_REQUESTED** [review]

**Review by DRadmir — APPROVED** [review]

**Pull request merged**